vuepress/docs/_posts/2023-03-17-building-business-based-autonomous-components.md

```diff
@@ -0,0 +1,192 @@
+
+Understanding the Bounded Contexts helps to design the PBCs you need to use or to develop. It is funny to see that both Hexagonal Architecture and PBCs are symbolized by a hexagonal shape, as they are both centered around the core business logic while they are concepts that are applied at different levels.
+
+On the other side, Composable Apps benefit PBCs structure by using them as bricks of business logic, making them independently composed and communicating using interfaces, and business-oriented contracts.
+
+Hexagonal Architecture or Composable Apps are following the same concepts but at a different level of abstraction using Domain Models as warrantors of the business core concepts.
+
+Does that mean Domain Centric Architecture is mandatory to use to build PBCs? Of course not, they both play at different stages of your application lifecycle. But, if you find Composable Apps useful to use to design your business application, there is no reason not to study these architectures to design your business code too. And, if you already use domain-centric practices to build your apps, using Composable Apps could be a good move to streamline these practices on the different layers of your business implementation.
+
+Back to our survey application, Bob can definitely help us understand the business context, that helps to code PBCs using a domain-centric architecture and to design the Composable Apps from a pack of these PBCs. Thanks to Bob, we would be able to share that knowledge through business components, share them with the company dev teams and let them reuse them.
+
+
+# Built for change
+
+Let’s admit it: we need experimentation to build great products. It is not possible to get it right on the first try (neither from a functional nor a technical perspective). So… if we have to make important changes many times, we have to build code that will welcome those changes!
+
+Types and Tests provides a pretty decent safety net to ensure that the solution is still working “as intended.” As we gain knowledge, the intent will change. New concepts will emerge, others will disappear. Code has to keep expressing that.
+
+With code built around a Domain Model, with really clear adaptations for the infrastructure parts, it is totally possible to refactor the whole thing without breaking anything! It is also possible to clearly extract a Bounded Context just by putting in a module (package in java, namespace in C#, …).
+
+Moving Bounded Contexts in modules and letting them live there for a while is a good plan since it’s way easier to do (or undo) than building a dedicated binary with its infrastructure.
```

Comment:
```suggestion
Moving bounded contexts into modules and letting them live there for a while is a good plan since it’s way easier to do (or undo) than building a dedicated binary with its infrastructure.
```